### PR TITLE
[Merged by Bors] - Added TlsAuthenticationProvider to be used in AuthenticationClass

### PR DESCRIPTION
**Pull request by maltesander**
## Description

- added TlsAuthenticationProvider as discussed in https://github.com/stackabletech/documentation/pull/186
- removed MutualTlsVerification

## Review Checklist
- [ ] Code contains useful comments
- [ ] (Integration-)Test cases added (or not applicable)
- [ ] Documentation added (or not applicable)
- [ ] Changelog updated (or not applicable)
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

File: src/commons/tls.rs [hunk 1]
```diff
@@ -43,3 +41,13 @@ pub enum CaCert {
     /// So if you got provided with a ca cert but don't have access to the key you can still use this method.
     SecretClass(String),
 }
+
+#[derive(Clone, Debug, Deserialize, Eq, JsonSchema, PartialEq, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct TlsAuthenticationProvider {
+    /// See `<https://github.com/stackabletech/documentation/tree/main/modules/contributor/pages/adr/ADR016-tls-authentication.adoc>`.
```
Comment (sbernauer) [inline src/commons/tls.rs:48]:
```suggestion
    /// See `<https://docs.stackable.tech/home/contributor/adr/ADR016-tls-authentication.html>`.
```

**Review by sbernauer — APPROVED** [review]
LGTM.
But I'm not sure if we should merge this prior to presenting the ADR on Wednesday...

**Comment by maltesander** [comment]
> LGTM. But I'm not sure if we should merge this prior to presenting the ADR on Wednesday...

Yeah we can wait for it, just would like to have it in the next release!

**Comment by maltesander** [comment]
bors merge

**Pull request closed without merging**